api/src/test/java/io/opencensus/metrics/MetricRegistryTest.java

```diff
@@ -33,153 +32,39 @@
-          @Override
-          public double applyAsDouble(Object value) {
-            return 5.0;
-          }
-        });
-  }
-
-  @Test
-  public void addDoubleGauge_NullFunction() {
-    thrown.expect(NullPointerException.class);
-    metricRegistry.addDoubleGauge(
-        "name", "description", "1", new LinkedHashMap<LabelKey, LabelValue>(), null, null);
-  }
-
   @Test
   public void addLongGauge_NullName() {
     thrown.expect(NullPointerException.class);
-    metricRegistry.addLongGauge(
-        null,
-        "description",
-        "1",
-        new LinkedHashMap<LabelKey, LabelValue>(),
-        null,
-        new ToLongFunction<Object>() {
-          @Override
-          public long applyAsLong(Object value) {
-            return 5;
-          }
-        });
+    metricRegistry.addLongGauge(null, "description", "1", new ArrayList<LabelKey>());
   }
 
   @Test
   public void addLongGauge_NullDescription() {
     thrown.expect(NullPointerException.class);
-    metricRegistry.addLongGauge(
-        "name",
-        null,
-        "1",
-        new LinkedHashMap<LabelKey, LabelValue>(),
-        null,
-        new ToLongFunction<Object>() {
-          @Override
-          public long applyAsLong(Object value) {
-            return 5;
-          }
-        });
+    metricRegistry.addLongGauge("name", null, "1", new ArrayList<LabelKey>());
   }
 
   @Test
   public void addLongGauge_NullUnit() {
     thrown.expect(NullPointerException.class);
-    metricRegistry.addLongGauge(
-        "name",
-        "description",
-        null,
-        new LinkedHashMap<LabelKey, LabelValue>(),
-        null,
-        new ToLongFunction<Object>() {
-          @Override
-          public long applyAsLong(Object value) {
-            return 5;
-          }
-        });
+    metricRegistry.addLongGauge("name", "description", null, new ArrayList<LabelKey>());
   }
 
   @Test
   public void addLongGauge_NullLabels() {
     thrown.expect(NullPointerException.class);
-    metricRegistry.addLongGauge(
-        "name",
-        "description",
-        "1",
-        null,
-        null,
-        new ToLongFunction<Object>() {
-          @Override
-          public long applyAsLong(Object value) {
-            return 5;
-          }
-        });
+    metricRegistry.addLongGauge("name", "description", "1", null);
   }
 
   @Test
-  public void addLongGauge_NullFunction() {
-    thrown.expect(NullPointerException.class);
-    metricRegistry.addLongGauge(
-        "name", "description", "1", new LinkedHashMap<LabelKey, LabelValue>(), null, null);
+  public void addLongGauge_NoopPoint() {
+    LongGauge longGaugeMetric =
+        metricRegistry.addLongGauge("name", "description", "1", new ArrayList<LabelKey>());
+    LongPoint dp = longGaugeMetric.getOrCreateTimeSeries(new ArrayList<LabelValue>());
+    dp.add(12);
+    dp.set(12);
+
+    longGaugeMetric.getDefaultTimeSeries();
+    longGaugeMetric.getOrCreateTimeSeries(new ArrayList<LabelValue>());
```

Comment:
We should have tests for remove (at least for null elements and the fact that it does not break if arguments are correct).